### DISCUSSION TITLE
loading both an image and label from the same file?

### DISCUSSION THREAD

**Discussion by tueboesen** [TorchIO-project/torchio#1283] [Q&A]
I have a bunch of data files where each datafile contains a tuple consisting of (sample, label).
Based on previous suggestions I was able to create the following to read in such torch files:
 ```

def torch_reader(path):
    sample,label = torch.load(path)
    return sample, None

subject_a = tio.Subject(
    data=tio.ScalarImage(file, reader=torch_reader),
)


```

However, the problem with this is that I also need to get the label from these datafiles.
Unfortunately there seems to be no way to pass back the label in this torch_reader, which does make sense since this is part of ScalarImage.

But nevertheless I still need that label. Pytorch does not allow me to just load part of a file as far as I'm aware, so I need some way to also load the label as I am loading the sample. However, I am not sure what the best way to do this might be, there seems to be no easy way to do this in the current framework.

It should be noted that my label is just a single number in this case, if that makes a difference.

**Comment by romainVala** [comment]
in that case I would suggest to directly construct your subject with something like that:

```
import torchio as tio
import numpy as np

sample,label = torch.load(path)
img = tio.ScalarImage(tensor=sample, affine = np.eye(4))
subject = tio.Subject(data=img, label=label) 

```
if your label is just a number then just add it in the subject, if it were an image, you could use tio.LabelMap
I hope it helps
